File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReachableDistance=nums[0];
        int i=1;
        while(maxReachableDistance>=i && i<nums.length){
            maxReachableDistance=Math.max(maxReachableDistance,nums[i]+i);
            i++;
        }
        return maxReachableDistance>=nums.length-1;
        
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) <= 1:
            return True
        j = len(nums) - 2
        i = len(nums) - 1    
        while j > -1:
            if j + nums[j] >= i:
                i = j
                j -= 1
                
            else:
                j -= 1
        #print('i', i, 'j', j)
        if i <= 0:
            return True
        else:
            return False